### DISCUSSION TITLE
Comfy UI nodes that renders/shows texts in HTML element other than `textarea`

### DISCUSSION THREAD

**Discussion by ilicnon** [comfyanonymous/ComfyUI#2988] [Q&A]
Does anyone have an example of comfy ui custom nodes that does this? The main reason i want to do this because almost all nodes ive seen that renders text uses `textarea` which apparently doesnt let you have custom color on pieces of text. Use case for this is that i will probably make a custom node that shows `booru formatted prompt` where each tags are seperately colored according to their tag groups (which are defined by danbooru), makes prompting easy (for me, probably) basically

**Comment by ilicnon** [comment]
![image](https://github.com/comfyanonymous/ComfyUI/assets/161084954/b54e0c1e-2c64-47a5-bd46-bf40fd2f47d3)
Something like this, probably

**Comment by tudoanh** [comment]
I also wanted this so bad. Please help.

**Comment by christian-byrne** [comment]
Use `node.addDOMWidget`:

```typescript
interface LGraphNode {
  addDOMWidget<
    T extends HTMLElement = HTMLElement,
    V extends object | string = string
  >(
    name: string,
    type: string, // INPUT_TYPES type
    element: T, // HTMLElement you can put anything inside of
    options?: {
      hideOnZoom?: boolean // whether to hide when zoomed out as with textarea widgets
      selectOn?: string[]
      onHide?: (widget: DOMWidget<T, V>) => void
      getValue?: () => V
      setValue?: (value: V) => void
      getMinHeight?: () => number
      getMaxHeight?: () => number
      getHeight?: () => string | number
      onDraw?: (widget: DOMWidget<T, V>) => void
      afterResize?: (this: DOMWidget<T, V>, node: LGraphNode) => void
    }
  ): DOMWidget<T, V>
}
```

**Reply by tudoanh** [reply]
Thank you so much!! This is working perfect.

**Reply by Pauan** [reply]
What about for Nodes V2?

**Reply by christian-byrne** [reply]
You can add a DOM widget in Nodes 2.0 as well.
